File: VB/WindowsApplication1/CustomEditor/MyButtonEditPainter.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraEditors.Drawing
Imports System.Drawing

Namespace WindowsApplication1
	Public Class MyButtonEditPainter
		Inherits ButtonEditPainter

		Public Sub New()

		End Sub

		Protected Overrides Sub DrawButton(ByVal viewInfo As DevExpress.XtraEditors.ViewInfo.ButtonEditViewInfo, ByVal info As EditorButtonObjectInfoArgs)
			If "IsImage".Equals(info.Button.Tag) Then
				Dim bounds As Rectangle = info.Bounds
				info.Graphics.FillRectangle(viewInfo.PaintAppearance.GetBackBrush(info.Cache), info.Bounds)
				bounds.Inflate(-3, -3)
				info.Graphics.DrawImage(info.Button.Image, bounds)
			Else
				MyBase.DrawButton(viewInfo, info)
			End If
		End Sub
	End Class
End Namespace

File: CS/WindowsApplication1/CustomEditor/MyButtonEditPainter.cs
using System;
using DevExpress.XtraEditors.Drawing;
using System.Drawing;

namespace WindowsApplication1
{
    public class MyButtonEditPainter : ButtonEditPainter
    {

        public MyButtonEditPainter()
        {

        }

        protected override void DrawButton(DevExpress.XtraEditors.ViewInfo.ButtonEditViewInfo viewInfo, EditorButtonObjectInfoArgs info)
        {
            if ("IsImage".Equals(info.Button.Tag))
            {
                Rectangle bounds = info.Bounds;
                info.Graphics.FillRectangle(viewInfo.PaintAppearance.GetBackBrush(info.Cache), info.Bounds);
                bounds.Inflate(-3, -3);
                info.Graphics.DrawImage(info.Button.Image, bounds);
            }
            else
                base.DrawButton(viewInfo, info);
        }
    }
}
